docs/index.rst

```diff
@@ -27,123 +27,316 @@ EdgeDB TypeScript/JS Client
+
+.. code-block:: bash
+
+  $ npx tsx seed.ts
+
+Feel free to explore the database in the EdgeDB UI, where you will find the new
+data you inserted through this script, as well as any data you inserted when
+running the Quickstart.
+
+.. note:: A word on module systems
+
+  Different build tools and runtimes have different specifications for how
+  modules are imported, and we support a wide-range of those styles. For
+  clarity, we will be sticking to standard TypeScript-style ESM module importing
+  without a file extension throughout this documentation. Please see your build
+  or environment tooling's guidance on how to adapt this style.
+
+Query
+^^^^^
+
+Now, let's write a Node.js script that queries the database for details about
+Iron Man 2:
+
+.. code-block:: typescript
+  :caption: query.ts
+
+  import * as edgedb from "edgedb";
+
+  const client = edgedb.createClient();
+
+  async function main() {
+    const result = await client.querySingle(`
+      select Movie {
+        title,
+        actors: {
+          name,
+        }
+      } filter .title = "Iron Man 2"
+    `);
+
+    console.log(JSON.stringify(result, null, 2));
+  }
+
+  main();
+
+Interfaces
+^^^^^^^^^^
+
+Since we're using TypeScript, it would be nice to be able to type the return
+value of this query, so let's use our first generator, the :ref:`Queries
+generator <edgedb-js-queries>` to tell TypeScript what the type of our result
+is.
+
+First we run the generator:
+
+.. code-block:: bash
+
+  $ npx @edgedb/generate interfaces
 
-  const edgedb = require("edgedb");
+This generator introspects your database schema and generates a set of
+equivalent TypeScript interfaces.
+
+Now we can annotate our query since we are selecting the whole ``Movie`` type:
+
+.. code-block:: typescript
+  :caption: query.ts
+
+  import * as edgedb from "edgedb";
+  import { Movie } from "./dbschema/interfaces"
 
   const client = edgedb.createClient();
-  const query = `select "Hello world!";`;
 
-  async function run(){
-    const result = await client.query(query)
-    console.log(result); // "Hello world!"
+  async function main() {
+    // result will be inferred as Movie | null
+    const result = await client.querySingle<Movie>(`
+      select Movie {
+        title,
+        actors: {
+          name,
+        }
+      } filter .title = "Iron Man 2"
+    `);
+
+    console.log(JSON.stringify(result, null, 2));
   }
 
-  run();
+  main();
 
-If you're not using TypeScript, you can skip straight to :ref:`the Client docs
-<edgedb-js-driver>`.
+You can now run the script with ``tsx``:
 
+.. code-block:: bash
 
-.. _edgedb-js-intro-qb:
+  $ npx tsx query.ts
 
-The query builder
-=================
+Queries generator
+^^^^^^^^^^^^^^^^^
+
+Wouldn't it be great if we could write any arbitrary query and get a type-safe
+function that we could call? Good news, that's exactly what the next generator
+does! The :ref:`queries generator <edgedb-js-queries>` Scans your project for
```

Comment:
Yeah, I think the names are sloppy in general:

1. Query Builder -> `generate edgeql-js`
2. Query Files -> `generate queries`
3. Interfaces -> `generate interfaces`

I think I like "Queries Generator" better than "Query Generator" since it generates zero or more "queries" instead of one "query".

So maybe the pages should be:

Query Builder Generator
Queries Generator
Interfaces Generator

WDYT?

Reply:
Perfect!